lib/bolt/task.rb

```diff
@@ -8,30 +8,33 @@ def initialize(target, task)
     end
   end
 
-  # Represents a Task.
-  # @file and @files are mutually exclusive.
-  # @name [String] name of the task
-  # @file [Hash, nil] containing `filename` and `file_content`
-  # @files [Array<Hash>] where each entry includes `name` and `path`
-  # @metadata [Hash] task metadata
-  Task = Struct.new(
-    :name,
-    :file,
-    :files,
-    :metadata
-  ) do
-    attr_reader :remote
-
-    def initialize(task, remote: false)
-      super(nil, nil, [], {})
-
+  class Task
+    METADATA_KEYS = %w[description extensions files implementations
```

Comment:
What is `extensions`? It's not listed in the [docs](https://puppet.com/docs/bolt/latest/writing_tasks.html#task-metadata-reference).

Reply:
It's reserved in the task spec as a place to put metadata that's specific to some task runner. It's not really used, but there might be a task or two out there that has some data under `extensions`.

https://github.com/puppetlabs/puppet-specifications/blob/master/tasks/README.md#options